src/app/clusters/scenes-server/SceneTableImpl.h

```diff
@@ -46,89 +46,110 @@ class DefaultSceneHandlerImpl : public scenes::SceneHandler
+        ReturnErrorOnFailure(app::DataModel::Encode(
+            writer, TLV::ContextTag(app::Clusters::Scenes::Structs::ExtensionFieldSet::Fields::kAttributeValueList), aVlist));
+        ReturnErrorOnFailure(writer.EndContainer(outer));
+        serializedBytes.reduce_size(writer.GetLengthWritten());
+
+        return CHIP_NO_ERROR;
+    }
+
+    /// @brief Decodes an attribute value list from a TLV structure and ensure it fits the member pair buffer
+    /// @param serializedBytes [in] Buffer to read from
+    /// @param aVlist [out] Attribute value list to fill from the TLV structure
+    /// @return CHIP_ERROR
+    virtual CHIP_ERROR DecodeAttributeValueList(
+        const ByteSpan & serializedBytes,
+        app::DataModel::DecodableList<app::Clusters::Scenes::Structs::AttributeValuePair::DecodableType> & aVlist)
+    {
+        TLV::TLVReader reader;
+        TLV::TLVType outer;
+        size_t pairTotal = 0;
+
+        reader.Init(serializedBytes);
+        ReturnErrorOnFailure(reader.Next(TLV::kTLVType_Structure, TLV::AnonymousTag()));
+        ReturnErrorOnFailure(reader.EnterContainer(outer));
+        ReturnErrorOnFailure(reader.Next(
+            TLV::kTLVType_Array, TLV::ContextTag(app::Clusters::Scenes::Structs::ExtensionFieldSet::Fields::kAttributeValueList)));
+        aVlist.Decode(reader);
+        ReturnErrorOnFailure(reader.ExitContainer(outer));
+        // Verify size of list
+        ReturnErrorOnFailure(aVlist.ComputeSize(&pairTotal));
+        VerifyOrReturnError(pairTotal <= ArraySize(mAVPairs), CHIP_ERROR_BUFFER_TOO_SMALL);
+
+        return CHIP_NO_ERROR;
+    }
+
     /// @brief From command AddScene, allows handler to filter through clusters in command to serialize only the supported ones.
     /// @param endpoint[in] Endpoint ID
     /// @param extensionFieldSet[in] ExtensionFieldSets provided by the AddScene Command, pre initialized
-    /// @param serialisedBytes[out] Buffer to fill from the ExtensionFieldSet in command
-    /// @return CHIP_NO_ERROR if successful, CHIP_ERROR_INVALID_ARGUMENT if the cluster is not supported, CHIP_ERROR value otherwise
+    /// @param serializedBytes[out] Buffer to fill from the ExtensionFieldSet in command
+    /// @return CHIP_NO_ERROR if successful, CHIP_ERROR_INVALID_ARGUMENT if the cluster is not supported, CHIP_ERROR value
+    /// otherwise
     virtual CHIP_ERROR SerializeAdd(EndpointId endpoint,
                                     const app::Clusters::Scenes::Structs::ExtensionFieldSet::DecodableType & extensionFieldSet,
                                     MutableByteSpan & serializedBytes) override
     {
-        app::Clusters::Scenes::Structs::AttributeValuePair::DecodableType aVPair;
-        TLV::TLVWriter writer;
-        TLV::TLVType outer;
-        size_t pairTotal  = 0;
-        uint8_t pairCount = 0;
+        app::Clusters::Scenes::Structs::AttributeValuePair::Type aVPairs[kMaxAvPair];
```

Comment:
Why the change to not use mAVPairs here?

Reply:
mAVPairs was originally introduced to ensure ExtensionFieldSet::Type.attributeValueList's pointer stays allocated after DefaultSceneHandlerImpl::Deserialize() returns. Since the data from aVPairs is all serialized into a buffer, I figured there is no need to share the mAVPairs buffer.